assigner/core/assigner_test.go

```diff
@@ -137,11 +136,10 @@ func TestAssignerAll(t *testing.T) {
 			}
 			t.Log("Publisher", peer2IDStr, "assigned to indexer", assignNum)
 			assigns = append(assigns, assignNum)
-		case <-timeout.C:
+		case <-time.After(3 * time.Second):
```

Comment:
What was the rationale for these changes? Would it be possible to use `require.Eventually` at the time of assertion to avoid fixed value waits?

Reply:
The rationale for this was because now (in go1.23) timers can be GC'd even if they are still running, so `time.After` means less code and no need to stop and reset timers and worry about semantics before and after go1.23.